pkg/reconciler/integrationsource/resources/containersource.go

```diff
@@ -0,0 +1,208 @@
+		switch field.Kind() {
+		case reflect.Int, reflect.Int32, reflect.Int64:
+			value = strconv.FormatInt(field.Int(), 10)
+		case reflect.Bool:
+			value = strconv.FormatBool(field.Bool())
+		case reflect.String:
+			value = field.String()
+		default:
+			// Skip unsupported types
+			continue
+		}
+
+		// Skip zero/empty values
+		if value == "" {
+			continue
+		}
+
+		envVars = append(envVars, corev1.EnvVar{
+			Name:  envVarName,
+			Value: value,
+		})
+	}
+
+	return envVars
+}
+
+// Function to create environment variables for Timer or AWS configurations dynamically
+func makeEnv(source *v1alpha1.IntegrationSource) []corev1.EnvVar {
+	var envVars []corev1.EnvVar
+
+	// Timer environment variables
+	if source.Spec.Timer != nil {
+		envVars = append(envVars, generateEnvVarsFromStruct("CAMEL_KAMELET_TIMER_SOURCE", *source.Spec.Timer)...)
+		return envVars
+	}
+
+	// Handle secret name only if AWS is configured
+	var secretName string
+	if source.Spec.Aws != nil && source.Spec.Aws.Auth != nil && source.Spec.Aws.Auth.Secret != nil && source.Spec.Aws.Auth.Secret.Ref != nil {
+		secretName = source.Spec.Aws.Auth.Secret.Ref.Name
+	}
+
+	// AWS S3 environment variables
+	if source.Spec.Aws != nil && source.Spec.Aws.S3 != nil {
+		envVars = append(envVars, generateEnvVarsFromStruct("CAMEL_KAMELET_AWS_S3_SOURCE", *source.Spec.Aws.S3)...)
+		if secretName != "" {
+			envVars = append(envVars, []corev1.EnvVar{
+				makeSecretEnvVar("CAMEL_KAMELET_AWS_S3_SOURCE_ACCESSKEY", "aws.s3.accessKey", secretName),
+				makeSecretEnvVar("CAMEL_KAMELET_AWS_S3_SOURCE_SECRETKEY", "aws.s3.secretKey", secretName),
+			}...)
+		}
+		return envVars
+	}
+
+	// AWS SQS environment variables
+	if source.Spec.Aws != nil && source.Spec.Aws.SQS != nil {
+		envVars = append(envVars, generateEnvVarsFromStruct("CAMEL_KAMELET_AWS_SQS_SOURCE", *source.Spec.Aws.SQS)...)
+		if secretName != "" {
+			envVars = append(envVars, []corev1.EnvVar{
+				makeSecretEnvVar("CAMEL_KAMELET_AWS_SQS_SOURCE_ACCESSKEY", "aws.s3.accessKey", secretName),
+				makeSecretEnvVar("CAMEL_KAMELET_AWS_SQS_SOURCE_SECRETKEY", "aws.s3.secretKey", secretName),
+			}...)
+		}
+		return envVars
+	}
+
+	// AWS DynamoDB Streams environment variables
+	if source.Spec.Aws != nil && source.Spec.Aws.DDBStreams != nil {
+		envVars = append(envVars, generateEnvVarsFromStruct("CAMEL_KAMELET_AWS_DDB_STREAMS_SOURCE", *source.Spec.Aws.DDBStreams)...)
+		if secretName != "" {
+			envVars = append(envVars, []corev1.EnvVar{
+				makeSecretEnvVar("CAMEL_KAMELET_AWS_DDB_STREAMS_SOURCE_ACCESSKEY", "aws.s3.accessKey", secretName),
+				makeSecretEnvVar("CAMEL_KAMELET_AWS_DDB_STREAMS_SOURCE_SECRETKEY", "aws.s3.secretKey", secretName),
+			}...)
+		}
+		return envVars
+	}
+
+	// If no valid configuration is found, return empty envVars
+	return envVars
+}
+
+func selectImage(source *v1alpha1.IntegrationSource) string {
+	if source.Spec.Timer != nil {
+		return "quay.io/openshift-knative/kn-connector-timer-source:1.0-SNAPSHOT"
```

Comment:
@dsimansk  for this to get rid of, let's see if we can get the integrations repo setup w/ image push etc.

and yes, this code will be changing too. but is first draft. I am currently more focused on having things work